oauth2_http/java/com/google/auth/oauth2/SecureSessionAgentConfig.java

```diff
@@ -33,7 +33,7 @@
 import com.google.errorprone.annotations.CanIgnoreReturnValue;
 
 /** Holds an mTLS configuration (consists of address of S2A) retrieved from the Metadata Server. */
-final class S2AConfig {
+public class SecureSessionAgentConfig {
```

Comment:
Can this continue to be final?

Reply:
final will let us mock in tests (if that's something we want to test in sdk PR). If not, I'm fine keeping this final